web/themes/custom/usagov/usagov.theme

```diff
@@ -10,13 +10,248 @@ use Drupal\taxonomy\Entity\Term;
+  $elements = $dom->documentElement->childNodes[0]->childNodes;
+  // Setup the first answer
+  $answer = new DOMDocument();
+  $answer->loadHTML("<html><body></body></html>");
+
+  // Count the number of H2s in the body content
+  $numH2s = count($dom->getElementsByTagName('h2'));
+
+  if ($numH2s > 1) {
+    // There are multiple H2s
+    // Each H2 is a question
+    $question = "No Questions Found Yet";
+
+    // Loop through elements of the body content
+    foreach ($elements as $element) {
+      // Check the type of element
+      if ($element->nodeName == "h2") {
+        // $element is a new question
+        // If there was an existing question, store it and the answer
+        if ($question != "No Questions Found Yet") {
+          $answer = faq_answer_inner_html($answer);
+          if ($question && $answer) {
+            $faq[] = (object) ['question' => $question, 'answer' => $answer];
+          }
+        }
+
+        // Set up the next question
+        $question = $element->nodeValue;
+
+        // Set up the next answer
+        $answer = new DOMDocument();
+        $answer->loadHTML("<html><body></body></html>");
+
+      }
+      elseif ($element->nodeName == "#text") {
+        // $element is a text node
+        // This is white space that we can ignore.
+      }
+      else {
+        // $element is some kind of content
+        // Append the content to the current answer
+        $importedNode = $answer->importNode($element, TRUE);
+        $answer->documentElement->childNodes[0]->appendChild($importedNode);
+      }
+    }
+
+    // If there was an existing question, store it and the answer
+    if ($question != "No Questions Found Yet") {
+      $answer = faq_answer_inner_html($answer);
+      if ($question && $answer) {
+        $faq[] = (object) ['question' => $question, 'answer' => $answer];
+      }
+    }
+
+  }
+  else {
+    // There are not multiple H2s
+    // The H1 is the question
+    $question = $node->getTitle();
+
+    // Append the intro to the answer
+    $introText = $node->get('field_page_intro')[0]->value;
+    if ($introText) {
+      $intro = $answer->createElement('p');
+      $intro->textContent = $introText;
+      $answer->documentElement->childNodes[0]->appendChild($intro);
+    }
+
+    // Loop through elements of the body content
+    foreach ($elements as $element) {
+      if ($element->nodeName == "#text") {
+        // $element is a text node
+        // This is white space that we can ignore.
+      }
+      else {
+        // $element is content for the answer
+        // Append the content to the current answer
+        $importedNode = $answer->importNode($element, TRUE);
+        $answer->documentElement->childNodes[0]->appendChild($importedNode);
+      }
+    }
+
+    $answer = faq_answer_inner_html($answer);
+    if ($question && $answer) {
+      $faq[] = (object) ['question' => $question, 'answer' => $answer];
+    }
+  }
+  $variables["dom"] = $dom;
```

Comment:
Please remove this line. I'll bet it's left over from debugging, but $variables isn't passed into this function, so it's not doing anything.

Reply:
Good catch!